oak_functions/loader/benches/lookup.rs

```diff
@@ -33,125 +31,199 @@ use oak_functions_loader::{
+                    &expected_value,
+                )
+            })
+        });
+    }
+    group.finish();
 }
 
-#[bench]
-fn medium_random_data_multi_lookup(bencher: &mut Bencher) {
-    let test_data = generate_random_test_data_for_bench(128, 256, 200_000);
+fn entry_count(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+    let wasm_module_bytes = bulid_wasm_module(MANIFEST_PATH);
 
-    run_lookup_bench(bencher, test_data, 101, Duration::from_millis(10));
-}
+    let mut group = c.benchmark_group("entry_count");
+    let plot_config = PlotConfiguration::default().summary_scale(AxisScale::Logarithmic);
+    group.plot_config(plot_config);
+    for count in [100usize, 10_000usize, 1_000_000usize].iter() {
+        let TestData {
+            test_key,
+            expected_value,
+            entries,
+        } = generate_random_test_data_for_bench(64, 256, *count);
+        let lookup_data = Arc::new(LookupData::for_test(entries.clone()));
+        let logger = Logger::for_test();
+        let wasm_handler = WasmHandler::create(&wasm_module_bytes, lookup_data, vec![], logger)
+            .expect("Couldn't create the server");
 
-#[bench]
-fn large_random_data_single_lookup(bencher: &mut Bencher) {
-    let test_data = generate_random_test_data_for_bench(1_024, 10_240, 1_000);
+        let single_benchmark_request = BenchmarkRequest {
+            iterations: SINGLE_REQUEST_ITERATIONS,
+            action: Some(Action::Lookup(LookupTest {
+                key: test_key.clone(),
+            })),
+        }
+        .encode_to_vec();
+        let multi_benchmark_request = BenchmarkRequest {
+            iterations: MULTI_REQUEST_ITERATIONS,
+            action: Some(Action::Lookup(LookupTest {
+                key: test_key.clone(),
+            })),
+        }
+        .encode_to_vec();
 
-    run_lookup_bench(bencher, test_data, 1, Duration::from_millis(5));
+        group.bench_with_input(BenchmarkId::new("single", count), count, |b, _size| {
+            b.iter(|| {
+                run_lookup_iteration(
+                    &rt,
+                    &wasm_handler,
+                    &single_benchmark_request,
+                    &expected_value,
+                )
+            })
+        });
+        group.bench_with_input(BenchmarkId::new("multi", count), count, |b, _size| {
+            b.iter(|| {
+                run_lookup_iteration(
+                    &rt,
+                    &wasm_handler,
+                    &multi_benchmark_request,
+                    &expected_value,
+                )
+            })
+        });
+    }
+    group.finish();
 }
 
-#[bench]
-fn large_random_data_multi_lookup(bencher: &mut Bencher) {
-    let test_data = generate_random_test_data_for_bench(1_024, 10_240, 1_000);
-
-    run_lookup_bench(bencher, test_data, 101, Duration::from_millis(25));
-}
+criterion_group!(benches, key_size, value_size, entry_count);
+criterion_main!(benches);
```

Comment:
What kind of output do we get from running the benches? In other benchmarks we have an expected bound on the execution time. Can we have a check like that here too, or is that not necessary?

Reply:
Running this produces output to stdout/stderr similar to:

```
key_size/multi/16       time:   [7.2419 ms 7.2496 ms 7.2587 ms]                              
Found 8 outliers among 100 measurements (8.00%)
  6 (6.00%) high mild
  2 (2.00%) high severe
```

It also generates an HTML report that contains charts of the performance at different points.

We cannot check that the execution time is below some bound, as benchmarking does not return a summary we can programatically inspect. That is why I moved the fixed lookup benchmark to the key_value_lookup example so that we can still check it somewhere.